c10/core/TensorImpl.cpp

```diff
@@ -584,26 +584,31 @@ void TensorImpl::Extend(int64_t num, float growthPct) {
   TORCH_CHECK(
       is_contiguous_,
       "Right now Extend is only supported for contiguous Tensor.");
+  TORCH_CHECK(
+      !has_symbolic_sizes_strides_,
+      "Extend() called on tensor with symbolic shape")
+
   using SizesVector = SmallVector<int64_t, 5>;
-  SizesVector newDims(
-      sizes_and_strides_.sizes_begin(), sizes_and_strides_.sizes_end());
+  IntArrayRef sizes_and_strides =
+      expectIntArrayRef(sizes_and_strides_.sizes_arrayref());
```

Comment:
expectIntArrayRef should be renamed to have "slow" in its name. Because you have tested `has_symbolic_sizes_strides_` an unchecked cast is OK, and we should have a way of spelling that too.